go/vt/sqlparser/sql.y

```diff
@@ -2331,6 +2370,7 @@ reserved_keyword:
 | LOCALTIMESTAMP
 | LOCK
 | MATCH
+| MAXVALUE
```

Comment:
I've previously misled people by pointing them at the mysql doc. It makes sense to follow the guidelines set by @bbeaudreault because we are a simplified grammar.

But make sure no conflicts are introduced in the grammar :).